library/general/block-wildcard-ingress/template.yaml

```diff
@@ -0,0 +1,33 @@
+apiVersion: templates.gatekeeper.sh/v1beta1
+kind: ConstraintTemplate
+metadata:
+  name: k8sblockwildcardingress
+  annotations:
+    description: >-
+      Users should not be able to create Ingresses with a blank or wildcard (*) hostname since that would enable them to intercept traffic for other services in the cluster, even if they don't have access to those services.
+spec:
+  crd:
+    spec:
+      names:
+        kind: K8sBlockWildcardIngress
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      # Try it out: https://play.openpolicyagent.org/p/Cfw0Ycx7OB
+      rego: |
+        package K8sBlockWildcardIngress
+
+        # This is how you express logical OR: https://www.openpolicyagent.org/docs/latest/#logical-or
+        is_wildcard = true {
+          object.get(input.review.object.spec.rules[_], "host", "") == ""
+        }
+        is_wildcard = true {
+          contains(object.get(input.review.object.spec.rules[_], "host", ""), "*")
+        }
+
+
+        violation[{"msg": msg}] {
+          input.review.kind.kind == "Ingress"
```

Comment:
would DRY things up a bit to have a variable like:

```
rules := input.review.object.spec.rules[_]
```

This would be used in the `hostename :=` line and could be passed into your two `is_wildcard` functions as a parameter.  This makes the code a bit easier to follow than its current state, where `input.review` is effectively accessed as a global variable.

Reply:
That ended up fixing the bug I mentioned where it was printing one message per host even when some of the hosts were valid :)

Reply:
Good catch!